src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts

```diff
@@ -1,24 +1,104 @@
-import { Pipe, PipeTransform } from '@angular/core';
+import { OnDestroy, Pipe, PipeTransform } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Subscription } from 'rxjs';
+
+@Pipe({
+    name: 'artemisDurationFromSeconds',
+    pure: false,
+})
+export class ArtemisDurationFromSecondsPipe implements PipeTransform, OnDestroy {
+    private readonly secondsInDay = 60 * 60 * 24;
+    private readonly secondsInHour = 60 * 60;
+    private readonly secondsInMinute = 60;
+
+    private seconds: number;
+    private onLangChange?: Subscription;
+
+    constructor(private translateService: TranslateService) {}
 
-@Pipe({ name: 'artemisDurationFromSeconds' })
-export class ArtemisDurationFromSecondsPipe implements PipeTransform {
     /**
-     * Convert seconds to a human-readable duration format (mm:ss).
+     * Convert seconds to a human-readable duration format:
+     * If shortenAndRound is false: "d day(s) hh:mm:ss", where the days and hours are left out if their value is zero
+     * If shortenAndRound is true: "xx unit yy unit", where only the two highest units are shown. (If the time is between 10 minutes and one hour, only the minutes are shown)
+     * @param short? {boolean} allows the format to be shortened
      * @param seconds {number}
      */
-    transform(seconds: number): string {
-        const minutes = Math.floor(seconds / 60);
-        seconds = seconds - minutes * 60;
+    transform(seconds: number, short?: boolean): string {
+        this.seconds = seconds;
+
+        const days = Math.floor(seconds / this.secondsInDay);
+        const hours = Math.floor((seconds % this.secondsInDay) / this.secondsInHour);
+        const minutes = Math.floor((seconds % this.secondsInHour) / this.secondsInMinute);
+        seconds = seconds % this.secondsInMinute;
 
-        let minutesOut = minutes.toString();
-        if (minutes < 10) {
-            minutesOut = '0' + minutes.toString();
+        if (short === true) {
+            return this.handleShortFormat(days, hours, minutes, seconds);
+        } else {
+            return this.handleLongFormat(days, hours, minutes, seconds);
         }
-        let secondsOut = seconds.toString();
-        if (seconds < 10) {
-            secondsOut = '0' + seconds.toString();
+    }
+
+    private handleShortFormat(days: number, hours: number, minutes: number, seconds: number): string {
+        if (days > 0) {
+            return days + 'd ' + hours + 'h';
+        } else if (hours > 0) {
+            return hours + 'h ' + minutes + 'min';
```

Comment:
It seems a bit odd to use 1d 1h 1min 1s and not m for minute. I know this was like that before, so it should not block merging, it just seems odd.

Reply:
'min' is commonly used for minutes, where other time units are often just abbreviated by a single letter.
I guess 'm' clashes with the Meter...